### PR TITLE
Issue 493 - First draft of data type description

### PR DESCRIPTION
**Pull request by kstreeter**
This PR addresses #493 by defining the logical type model for XDM, and adding a new `meta:xdmType` attribute to the meta schema.

### CODE REVIEW

File: docs/data_types.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## Data Type Definitions
+
+The following table describes the available data types:
+
+| XDM type  | Range                                                     | JSON Schema type | JSON Schema format |
+| --------- | --------------------------------------------------------- | ---------------- | ------------------ |
+| string    | unlimited                                                 | string           |                    |
+| number    | ±2.23×10^308 to ±1.80×10^308 (IEEE 64-bit floating point) | number           |                    |
+| long      | (-2^53 + 1) - (2^53 + 1) (54-bit signed integer\*)        | integer          |                    |
```
[the remark this answers is not in the document]
Comment (kstreeter) [reply docs/data_types.md]: @lrosenthol, are you saying that we should just call the range as 64-bit, leave the note that that in practice its only 53-bit?

I feel like we should just acknowledge the limitation here.  The JSON/Javascript ecosystem only really supports a 54-bit integer.  We want to interop with JS, so we adopt the same.

File: docs/data_types.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+| int       | -2^31 - 2^31 (32-bit signed integer)                      | integer          |                    |
+| short     | -2^15 - 2^15 (16-bit signed integer)                      | integer          |                    |
+| byte      | -2^7 - 2^7 (8-bit signed integer)                         | integer          |                    |
+| boolean   | { true, false }                                           | boolean          |                    |
+| date      | n/a                                                       | string           | date               |
```
[the remark this answers is not in the document]
Comment (kstreeter) [reply docs/data_types.md]: This is fixed in the latest commit

**Comment by lrosenthol** [comment]
@cdegroot-adobe - we have previously agreed that while the original JSON design (and the name) was built around JavaScript, neither JSON-LD nor JSON-Schema is built on JavaScript or its limitations.

That said, provided that we can resolve the `xdmType` issue, then I am OK with putting this limitation on XDM's long and saying that if you need something larger you must use BigNum - but that means we finalize that piece of work at the same time.

File: docs/data_types.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kstreeter) [reply docs/data_types.md]: There is an error here @cdegroot-adobe, although the range you suggested is not correct either.  From RFC 8259 section 6:

> Note that when such software is used, numbers that are integers and
   are in the range [-(2**53)+1, (2**53)-1] are interoperable in the
   sense that implementations will agree exactly on their numeric
   values.

**Comment by kstreeter** [comment]
Thanks @lrosenthol.  I filed issue #519 to track the need for BigDecimal/BigInteger types in XDM.

**Pull request merged**